Core/include/Acts/Geometry/NavigationDelegates.hpp

```diff
@@ -0,0 +1,85 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/NavigationState.hpp"
+#include "Acts/Utilities/Delegate.hpp"
+
+#include <memory>
+#include <tuple>
+
+/// @note this is foreseen for the 'Geometry' module
+
+namespace Acts {
+
+class Surface;
+
+namespace Experimental {
+
+class Portal;
+class DetectorVolume;
+class Detector;
+
+/// Base class for navigaiton delegate implementations
+/// that allows to then do some sort of type erasure,
+/// e.g. for plotting or I/O.
+///
+/// The interface is never touched in the navigation stream
+class IDelegateImpl {
```

Comment:
I had some problems with that, and I eventually wanted to require a `name()` method in order to give at least some screen info ... ?

Reply:
Hmm not sure what the reasoning there is, but I would be very carefully with using void pointers anywhere, because then we completely loose type safety...

Reply:
Ok, let's keep as it.

Reply:
I take back the point, I have misunderstood what was meant by void pointer in this case...

Reply:
To answer Paul:
- we do/will do downcasts into the concrete types (the code is not part of this PR), e.g. when displaying the relationship of surfaces inside a grid, or when writing the DetectorVolume to Json (will be part of a follow-up PR).

- however, we never need to cast to `IDelegateImpl`, we can directly do the cast sequence into the possible implementations